rfd/00XX-ci.md

```diff
@@ -0,0 +1,83 @@
+
+This includes, but is not limited to, the following functionality.
+
+* Automating parts of the development process (approvals, assignments) with bots.
+* Running tests (unit, integration) against internal and external PRs.
+* Running scheduled workflows (nightly) against `master`.
+* Tracking and reporting metrics (test failures, regressions) to external tooling (Slack).
+
+## Details
+
+### Security
+
+To achieve these goals, the below is proposed.
+
+* Continuous Integration (CI) and Continuous Delivery (CD) should be separate systems. Exploiting CI should not lead to a compromise of CD.
+* We should not self-host CI, instead we should buy a CI service we trust. GitHub Actions is proposed in the RFD.
+* User controlled input must always run in an unprivileged context.
+* All business logic should be performed a small Go program called `workflow-bot`. This will let us write test coverage in the majority of the CI workflow.
+
+To maintain the above security property, the following changes will be made.
+
+#### Repository Settings
+
+`CODEOWNERS` will be updated to only protect the [`.github/workflows` directory](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners). Any changes to GitHub Workflows will require repository administrators (like @klizhentas, @russjones, or @r0mant) to review and approve. Approvals for all other directories will be covered in `RFD 00XX: Assignment and Approval Workflows`.
+
+Branch protection rules will be updated to [dismiss approvals](https://docs.github.com/en/github/administering-a-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule#creating-a-branch-protection-rule) to `master` and all release branches if new changes are pushed to the PR after approval. To prevent a slowdown to developer productivity, `Require branches to be up to date before merging` will be disabled. Any breaking changes caused by this will be caught by automated testing that will run against `master` and all release branches upon merge.
+
+#### Workflow Permissions
+
+Workflows will run either in the context of `master` or `HEAD` of the PR and [with access to secrets or without](https://docs.github.com/en/actions/reference/encrypted-secrets#accessing-your-secrets). Use cases are illustrated in the `Use Cases and Permissions` section.
+
+### Use Cases and Permissions
+
+#### PR Approval Bot
+
+Context: `master`, Secrets: `GITHUB_TOKEN: READ`
+
+A bot that approves PRs should not run in the context of the PR, otherwise an attacker could simply change the approval workflow itself, and approve their own PR. A PR approval bot should run in the context of `master`.
+
+In addition, because an approval bot does not need to make changes to the repository, it should not have access to any additional secrets and should only have `GITHUB_TOKEN: READ` permissions.
```

Comment:
It isn't clear to me that `READ` and `WRITE` cover the necessary scope of permissions.  Doesn't this workflow need to:

```
POST /repos/gravitational/teleport/pulls/7204/reviews
POST /repos/gravitational/teleport/pulls/7402/reviews/XXX/events
```
https://docs.github.com/en/rest/reference/pulls#create-a-review-for-a-pull-request
https://docs.github.com/en/rest/reference/pulls#submit-a-review-for-a-pull-request

Is that implicit in the above token?